Calculator.py

```diff
@@ -0,0 +1,134 @@
+import re
+import math
+from statistics import mean
+
+from pip._vendor.msgpack.fallback import xrange
+
+
+class Calculator:
+    def __init__(self, exercise):
+        self.__operations_regex_dict = {1: r'\(\d+[!%^+/*\-~$&@]\d+\)+',
+                                        2: r'\!',
+                                        3: r'[$&@]',
+                                        4: r'%',
+                                        5: r'\^',
+                                        6: r'[/*]',
+                                        7: r'[-+]'}
+        self.exercise = exercise.strip()
+        self.solved = ''
+        print(self.arrange)
+
+    @property
+    def arrange(self):
+        self.solved = re.search(r'(!\d+)|(([~!]\d+)?\d+[-()%^/*+$&@]([~!]\d+)?\d+)', self.exercise)
+        if self.solved is None:
+            return self.exercise
+        exercise = self.exercise
+        for operation in self.__operations_regex_dict.values():

```

Comment:
your logic here is a bit all over the place. Your goal (if I understand correctly) is to handle each operation in order of its priority. However what you seem to be doing  is handling each operation in random order which is a problem.

Feel free to correct me if I am misunderstanding but there is definitely something here that is at a minimum done strangely.